risc0/circuit/rv32im/src/prove/emu/exec/mod.rs

```diff
@@ -590,10 +593,10 @@ impl<'a, 'b, S: Syscall> Executor<'a, 'b, S> {
     }
 
     fn ecall_bigint2(&mut self) -> Result<bool> {
-        let blob_ptr = self.load_guest_addr_from_register(REG_A0)?.waddr();
-        let nondet_program_ptr = self.load_guest_addr_from_register(REG_T1)?;
-        let verify_program_ptr = self.load_guest_addr_from_register(REG_T2)?;
-        let consts_ptr = self.load_guest_addr_from_register(REG_T3)?;
+        let blob_ptr = self.load_aligned_guest_addr_from_register(REG_A0)?.waddr();
+        let nondet_program_ptr = self.load_aligned_guest_addr_from_register(REG_T1)?;
+        let verify_program_ptr = self.load_aligned_guest_addr_from_register(REG_T2)?;
+        let consts_ptr = self.load_aligned_guest_addr_from_register(REG_T3)?;
```

Comment:
I'm not 100% certain that bigint2 addresses need to be aligned, but my understanding of how it works would imply they probably do. If someone could check my understanding here, that would be helpful.

Reply:
cc @jbruestle @tzerrell

Reply:
Technically, only the `verify_program_ptr` needs to be aligned since that is the pointer that is dereferenced by the circuit. The rest of these pointers are only loaded by the executor/preflight, so those should be safe to be unaligned. It probably doesn't hurt to check all of these.

Reply:
That makes sense. It seems like it doesn't make a huge difference either way then. I'd lean towards checking alignment here.